boa_engine/src/module/source.rs

```diff
@@ -1600,22 +1600,23 @@ impl SourceTextModule {
         envs.push_module(env);
 
         // 9. Set the Function of moduleContext to null.
+        // 10. Assert: module.[[Realm]] is not undefined.
+        // 11. Set the Realm of moduleContext to module.[[Realm]].
         // 12. Set the ScriptOrModule of moduleContext to module.
+        // 13. Set the VariableEnvironment of moduleContext to module.[[Environment]].
+        // 14. Set the LexicalEnvironment of moduleContext to module.[[Environment]].
+        // 15. Set the PrivateEnvironment of moduleContext to null.
         let call_frame = CallFrame::new(
             codeblock.clone(),
             Some(ActiveRunnable::Module(parent.clone())),
-            None,
+            envs,
+            realm.clone(),
         );
         context.vm.push_frame(call_frame);
```

Comment:
Thought: Would it be worthwhile adding an extra method on `vm` akin to `push_frame_with_stack` or something similar.

There's a few times where the frame is pushed and then the this and function value are pushed onto the stack separate. Is there any value in coupling the frame push and the stack push in these instances.

Reply:
> Thought: Would it be worthwhile adding an extra method on `vm` akin to `push_frame_with_stack` or something similar.
>
> There's a few times where the frame is pushed and then the this and function value are pushed onto the stack separate. Is there any value in coupling the frame push and the stack push in these instances.

Having a utility function seems fine, maybe even `pop_frame_with_stack`, Ill add it :)